src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonConverter.cs

```diff
@@ -54,6 +54,11 @@ internal ConverterStrategy ConverterStrategy
         /// </summary>
         internal virtual bool SupportsCreateObjectDelegate => false;
 
+        /// <summary>
+        /// Indicates that the converter is compatible with <see cref="JsonObjectCreationHandling.Populate"/>.
+        /// </summary>
+        internal virtual bool CanPopulate => false;
```

Comment:
we use effective converter when we check this value at the property so this setting will be ignored for factories

Reply:
Can `CanPopulate` be computed using `IsInternalConverter` and `TypeInfoKind`?

Reply:
no. For collections/dictionary it's really up to collection/dictionary converter if it does support it or not. I'm also thinking about future "AdvancedConverter" which would ideally have a way to support this as well. For object converters it technically could be just checked with TypeInfoKind since they always support it.